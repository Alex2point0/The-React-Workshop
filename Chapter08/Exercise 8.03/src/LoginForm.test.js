import React from 'react';
import {render, fireEvent, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import {MyEnhancedLoginForm} from './WithFormikExample';

test('renders correctly', () => {
  const {container} = render(<MyEnhancedLoginForm/>);
  expect(container).toMatchSnapshot();
});

test('loads and displays form', async () => {
  //Arrange
  const {container} = render(<MyEnhancedLoginForm/>);
  const email = container.querySelector('input[type="text"]');
  const password = container.querySelector('input[type="password"]');

  //Act
  await wait(() => {
    fireEvent.change(email, {
      target: {
        value: 'example@email.com'
      }
    });
    fireEvent.change(password, {
      target: {
        value: 'password'
      }
    })
  });

  //Assert
  expect(email.value).toBe('example@email.com');
  expect(password.value).toBe('password');
});

test('submits the form', async () => {
  //Arrange
  const {container} = render(<MyEnhancedLoginForm/>);
  const email = container.querySelector('input[type="text"]');
  const password = container.querySelector('input[type="password"]');
  const submit = container.querySelector('input[type="submit"]');

  expect(submit.disabled).toBeFalsy();

  //Act
  await wait(() => {
    fireEvent.change(email, {
      target: {
        value: 'example@email.com'
      }
    });
    fireEvent.change(password, {
      target: {
        value: 'password'
      }
    })
  });

  await wait(() => {
    fireEvent.click(submit);
  });

  expect(submit.disabled).toBeTruthy();
});

