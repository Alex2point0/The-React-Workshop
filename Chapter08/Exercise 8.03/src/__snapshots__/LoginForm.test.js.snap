// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form>
    <label>
      Name:
      <input
        name="name"
        type="text"
        value=""
      />
    </label>
    <label>
      Password:
      <input
        name="password"
        type="password"
        value=""
      />
    </label>
    false
    <input
      type="submit"
      value="Login"
    />
    Status: 
    Is Validating
  </form>
</div>
`;
